pkg/util/templates/templates.go

```diff
@@ -55,3 +58,129 @@ func TemplateForServiceMonitorResource(url string, namespacedName types.Namespac
 	}
 	return serviceMonitor
 }
+
+// TemplateForPrometheusRuleResource returns a PrometheusRule
+func TemplateForPrometheusRuleResource(url, percent string, namespacedName types.NamespacedName) monitoringv1.PrometheusRule {
+
+	routeURL := url
+	routeURLLabel := fmt.Sprintf(`RouteMonitorUrl="%s"`, routeURL)
+	rules := []monitoringv1.Rule{}
+
+	for _, thunderStruct := range []struct {
+		duration  string
+		severity  string
+		timeShort string
+		timeLong  string
+	}{
+		{
+			duration:  "2m",
+			severity:  "critical",
+			timeLong:  "1h",
+			timeShort: "5m",
+		},
+		{
+			duration:  "15m",
+			severity:  "critical",
+			timeLong:  "6h",
+			timeShort: "30m",
+		},
+		{
+			duration:  "1h",
+			severity:  "warning",
+			timeLong:  "1d",
+			timeShort: "2h",
+		},
+		{
+			duration:  "3h",
+			severity:  "warning",
+			timeLong:  "3d",
+			timeShort: "6h",
+		},
+	} {
+		// Create all the alerts
+		// Sample resource
+		/*
+		   - alert: ErrorBudgetBurn
+		     annotations:
+		       message: 'High error budget burn for targeturl=getmeright (current value: {{ $value }})'
+		     expr: |
+		       sum(http_requests_total:burnrate5m{targeturl="getmeright"}) > (14.40 * (1-0.95000))
+		       and
+		       sum(http_requests_total:burnrate1h{targeturl="getmeright"}) > (14.40 * (1-0.95000))
+		     for: 2m
+		     labels:
+		       severity: critical
+		       targeturl: getmeright
+		*/
+
+		const alertTemplate string = `
+	        	 sum(http_requests_total:burnrate%[3]s{%[1]s}) > (14.40 * (1-%[2]s))
+	        	 and
+	        	 sum(http_requests_total:burnrate%[4]s{%[1]s}) > (14.40 * (1-%[2]s))
+			 `
+		rules = append(rules, monitoringv1.Rule{
+			Alert: "ErrorBudgetBurn",
+			Expr: intstr.FromString(fmt.Sprintf(alertTemplate,
+				routeURLLabel,
+				percent,
+				thunderStruct.timeShort,
+				thunderStruct.timeLong)),
+			Labels: sampleTemplateLabelsWithSev(routeURL, thunderStruct.severity),
+			Annotations: map[string]string{
+				"message": fmt.Sprintf("High error budget burn for %s (current value: {{ $value }})", routeURLLabel),
+			},
+			For: thunderStruct.duration,
+		})
+	}
+
+	// Create all the recording rules
+	// Sample resource
+	/*
+	   - expr: |
+	       sum(rate(http_requests_total{targeturl="getmeright",code=~"5.."}[6h]))
+	       /
+	       sum(rate(http_requests_total{targeturl="getmeright"}[6h]))
+	     labels:
+	       targeturl: getmeright
+	     record: http_requests_total:burnrate6h
+	*/
+	const recordingRuleTemplate string = `
```

Comment:
that way it's going to contain all the tab and whitespace characters, is that wanted?

Reply:
same as before